docs/IE_PLUGIN_DG/CompiledModel.md

```diff
@@ -0,0 +1,87 @@
+# Compiled Model {#openvino_docs_ov_plugin_dg_compiled_model}
+
+`CompiledModel` class functionality:
+- Compile an ov::Model instance to a backend specific graph representation
+- Create an arbitrary number of `InferRequest` objects
+- Hold some common resources shared between different instances of `InferRequest`. For example:
+	- ov::ICompiledModel::m_task_executor task executor to implement asynchronous execution
+	- ov::ICompiledModel::m_callback_executor task executor to run an asynchronous inference request callback in a separate thread
+
+`CompiledModel` Class
+------------------------
+
+OpenVINO Plugin API provides the interface ov::ICompiledModel which should be used as a base class for a compiled model. Based on that, a declaration of an compiled model class can look as follows: 
+
+@snippet src/compiled_model.hpp compiled_model:header
+
+### Class Fields
```

Comment:
maybe we need to rename `Class` to `Class declaration` and keep it on the same level as all other headings below? It will allow users to see the whole table of content
<img width="191" alt="Screenshot 2023-03-16 at 1 32 18 PM" src="https://user-images.githubusercontent.com/2566854/225575090-d74f2191-3d4e-424a-8306-72fef8ef734d.png">


If yes, we can do it from `Plugin` as well.

Reply:
Fixed